### ISSUE TITLE
Emulate exceptions / memory faults

### ISSUE THREAD

**Issue by TheGag96** [citra-emu/citra#6874]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

Citra should somehow emulate throwing exceptions on memory faults (especially reads/writes to null pointers / the null page) that can be easy inspected with the GDB stub.

### Why would this feature be useful?

While developing some homebrew, I've noticed that Citra happily chugs along if my program reads/writes in the null page, only leaving an error log if I've ran my program with `citra`. This can be really nasty because I can see my program basically working in Citra but be in for a surprise trying to run it on a real 3DS. By that point, I don't have a way to get a good stack trace (short of writing my own crash handler I guess), and so hopefully using the PC address in the crash dump to point me to the line where it crashed will be enough to lead me in the right direction. If I could trap on the exception in Citra, it would save a lot of time and guesswork.

Thanks.

**Comment by wheremyfoodat** [comment]
This is not a perfect replacement for what you're asking for by any means, but I believe you should also be getting an error in your logs whenever you get an invalid memory access. But yeah, a proper way to trap data aborts is preferable.

https://github.com/citra-emu/citra/blob/6d0cd5b00e80bdea60d7e0e6caed81f7cd5fbd40/src/core/memory.cpp#L471-L474

**Comment by TheGag96** [comment]
Yes I have seen those - in many cases it hasn't been helpful because it happens in a common function at the bottom of a callstack, unfortunately.
